### PR TITLE
Add MD5 hash for x/crypto

### PR DESCRIPTION
**Pull request by notch1p**
also includes a fix for sha1. see #22

### CODE REVIEW

File: crypto/md5_test.mbt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  let a = Array::make(1000,0)
+  for index = 0; index < 1000; index = index + 1 {
+    a[index] = index % 256
+  }
+
+  let hash = @crypto.md5(Bytes::from_array(a))
+  inspect!(
+    bytes_to_hex_string(hash),
+    content="cbecbdb0fdd5cec1e242493b6008cc79"
+  )
```
Comment (peter-jerry-ye) [inline crypto/md5_test.mbt:55]: Not a very good test case because it's hard to verify with other tools

File: crypto/sha1.mbt [hunk 2]
```diff
@@ -20,7 +20,7 @@ pub fn sha1(input : Bytes) -> Bytes {
   if new_length - old_length < 9 {
     new_length += 64
   }
-  let bytes = Bytes::make(new_length)
+  let bytes = Bytes::make(new_length,b'\x00')
```
Comment (peter-jerry-ye) [inline crypto/sha1.mbt:23]: Execute `moon fmt`

File: crypto/md5_test.mbt (hunk 1 above)
Comment (notch1p) [reply crypto/md5_test.mbt:55]: This is copied from sha1 test, should I remove that as well?

File: crypto/md5.mbt [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+// An MD5 implementation based on
+// [RFC1321]       https://www.ietf.org/rfc/rfc1321.txt
+// [Ron Rivest]    https://people.csail.mit.edu/rivest/Md5.c
+// [md5-0.7.0]     https://docs.rs/md5/0.7.0/src/md5/lib.rs.html
+//
+//
+//! Security Notice
+//! MD5 is considered cryptographically broken and unsuitable for further use. 
+//! Collision attacks against MD5 are both practical and trivial, 
+//! and theoretical attacks against MD5 have been found.
```
[the remark this answers is not in the document]
Comment (notch1p) [reply crypto/md5.mbt:25]: could do like this
```moonbit
//! Security Notice
//! MD5 is considered cryptographically broken and unsuitable for further use. 
//! Collision attacks against MD5 are both practical and trivial, 
//! and theoretical attacks against MD5 have been found.
// - [RFC1321]       https://www.ietf.org/rfc/rfc1321.txt
// - [Ron Rivest]    https://people.csail.mit.edu/rivest/Md5.c
// - [md5-0.7.0]     https://docs.rs/md5/0.7.0/src/md5/lib.rs.html 
/// Compute the MD5 digest of some `data` based on RFC1321
pub fn md5(data : Bytes) -> Bytes {
  let ctx = MD5Context::make()
  md5_update(ctx, data)
  ctx.compute()
}
```

File: crypto/README.md [hunk 4]
```diff
@@ -6,9 +6,18 @@ A collection of cryptographic hash functions and utilities.
...
 ## Usage
 
+> Strings in MoonBit are UTF-16 LE encoded.
+
 ### SHA-1
 
 ```moonbit
 let input = "The quick brown fox jumps over the lazy dog"
-println(bytes_to_hex_string(sha1(a.to_bytes()))) // bd136cb58899c93173c33a90dde95ead0d0cf6df
+bytes_to_hex_string(sha1(input.to_bytes())) // bd136cb58899c93173c33a90dde95ead0d0cf6df
```
[the remark this answers is not in the document]
Comment (notch1p) [reply crypto/README.md:15]: Now changed to this
```moonbit
let input = "The quick brown fox jumps over the lazy dog"
println(bytes_to_hex_string(md5(input.to_bytes()))) // b0986ae6ee1eefee8a4a399090126837
```

File: crypto/md5.mbt [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+  let a = Ref::new(state[0])
+  let b = Ref::new(state[1])
+  let c = Ref::new(state[2])
+  let d = Ref::new(state[3])
+  fn t(
+    op : (UInt, UInt, UInt) -> UInt,
+    _a : Ref[UInt],
+    _b : Ref[UInt],
+    _c : Ref[UInt],
+    _d : Ref[UInt],
```
[the remark this answers is not in the document]
Comment (notch1p) [reply crypto/md5.mbt:146]: Now it's renamed to `state_{a,b,c,d}`

File: crypto/md5_test.mbt (hunk 1 above)
Comment (notch1p) [reply crypto/md5_test.mbt:55]: sha1_test has multiple tests like this:
https://github.com/moonbitlang/x/blob/d99105ac8d12f5cb17950a1cc79551df3e137e26/crypto/sha1_test.mbt#L21-L60
It is indeed confusing, I've verified using python's hashlib, which is not convenient at all

Comment (peter-jerry-ye) [reply crypto/md5_test.mbt:55]: Well, I think you can leave it like this.

**Pull request merged**